<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <title>MacroHapDB</title>
  <script>
    function readTable() {
      var file = document.getElementById("file").files[0];
      var reader = new FileReader();
      reader.onload = function(event) {
        var contents = event.target.result;
        var lines = contents.split("\n");
        var uniqueValues = new Set();
        for (var i = 0; i < lines.length; i++) {
          var cells = lines[i].split("\t");
          uniqueValues.add(cells[0]);
        }
        var select = document.getElementById("select");
        var sortedValues = Array.from(uniqueValues).sort();
        for (let value of sortedValues) {
          var option = document.createElement("option");
          option.text = value;
          select.add(option);
        }
        showRows();
      };
      reader.readAsText(file);
    }

    function showRows() {
      var file = document.getElementById("file").files[0];
      var reader = new FileReader();
      reader.onload = function(event) {
        var contents = event.target.result;
        var lines = contents.split("\n");
        var selectedValue = document.getElementById("select").value;
        var table = "<table>";
        for (var i = 0; i < lines.length; i++) {
          var cells = lines[i].split("\t");
          if (cells[0] == selectedValue) {
            table += "<tr>";
            for (var j = 0; j < cells.length; j++) {
              table += "<td>" + cells[j] + "</td>";
            }
            table += "</tr>";
          }
        }
        table += "</table>";
        document.getElementById("output").innerHTML = table;
      };
      reader.readAsText(file);
    }
  </script>
</head>
<body>
  <h1>Macrohap database (MacroHapDB) v1.0 </h1/>
  <p><br>
  A Macrohap Allele consists of an alleic sequence at variant sites only of forensic CODIS STRs, SNPs and InDels.  MacroHapDB deposits alleles,cross-validated information and their frequency count  in human populations. The current version includes two sets of humman samples, each with three databases at population, supperpopulation and global scale, from the 1000 genome project.<br>
  Dataset I: 289 validated individuals; 11,560 alleles.
  Dataset II: 2504 individuals; 100,160 alleles.
  Both have 26 populations (FIN,CDX,IBS,MXL,CHB,CHS,CEU,JPT,ESN,KHV,TSI,CLM,YRI,GBR,PEL,STU,BEB,GIH,PJL,MSL,ITU,GWD,LWK,ASW,PUR,ACB), 5 supperpopulations (EAS, EUR, AFR,SAS, AMR).  </p>
  
  <p>Usage: <br>
  Step 1. choose a database file: .global, pop, superPop <br>
  Step 2. select a marker name from drop-down list <br>
  Step 3. the alleles for each marker will be shown up <br>
  Contact: Xuewen.Wang at UNTHSC.edu, August.Woerner at UNTHSC.edu</p>

  <input type="file" id="file" name="file" accept=".txt" onchange="readTable()" value="default.txt">
  <br>
  <label for="select">Select a marker name:</label>
  <select id="select" onchange="showRows()">
    <option value="" selected disabled hidden style='display: none;'>None</option>
  </select>
  <br>
  <hr>
  <div id="output"></div>
</body>
</html>
